modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLViewEntriesDisplayContext.java

```diff
@@ -366,6 +368,17 @@ public boolean isVersioningStrategyOverridable() {
 		return _dlAdminDisplayContext.isVersioningStrategyOverridable();
 	}
 
+	private String _addDoAsUserIdParameter(String url) {
```

Comment:
Hey @marcogalluzzi 

_portal.addPreservedParameters method does almost the same (and has an specific parameter to control the doAsUserId parameter).

In order to avoid the logic duplication why don't we use this method directly instead of creating _addDoAsUserIdParameter in 2 places?

Thanks!

Reply:
You are right, and I don't like reapeating code. The only problem I see is that `_portal.addPreservedParameters` can also add the parameter `DoAsUserLanguageId` which is not needed in the cases I'm trying to fix.

Reply:
MMM, if both parameters are appended together maybe it could be necessary in some cases as well...

I mean, logic now seems ok and in my opinion is in the right place so I'd stop this for this detail. 

If adolfo agrees and passes it is good to go for me.

👍